node/src/main.rs

```diff
@@ -866,70 +880,187 @@ async fn main() {
-            {
-                return Err(format_err!(
-                    "Is your Ethereum node string missing a network name? \
+    networks.fold(Ok(EthereumNetworks::new()), |networks, network_arg| {
+        if network_arg.starts_with("wss://")
+            || network_arg.starts_with("http://")
+            || network_arg.starts_with("https://")
+        {
+            return Err(format_err!(
+                "Is your Ethereum node string missing a network name? \
                      Try 'mainnet:' + the Ethereum node URL."
-                ));
-            } else {
-                // Parse string (format is "NETWORK_NAME:URL")
-                let split_at = network.find(':').ok_or_else(|| {
-                    return format_err!(
-                        "A network name must be provided alongside the \
+            ));
+        } else {
+            // Parse string (format is "NETWORK_NAME:NETWORK_CAPABILITIES:URL" OR
+            // "NETWORK_NAME::URL" which will default to NETWORK_CAPABILITIES="archive,traces")
+            let split_at = network_arg.find(':').ok_or_else(|| {
+                return format_err!(
+                    "A network name must be provided alongside the \
                          Ethereum node location. Try e.g. 'mainnet:URL'."
-                    );
-                })?;
+                );
+            })?;
 
-                let (name, loc_with_delim) = network.split_at(split_at);
-                let loc = &loc_with_delim[1..];
+            let (name, rest_with_delim) = network_arg.split_at(split_at);
+            let rest = &rest_with_delim[1..];
+            if name.is_empty() {
+                return Err(format_err!(
+                    "Ethereum network name cannot be an empty string"
+                ));
+            }
 
-                if name.is_empty() {
-                    return Err(format_err!(
-                        "Ethereum network name cannot be an empty string"
-                    ));
-                }
+            let http_split_at = rest.find("http").ok_or_else(|| {
+                return format_err!(
+                    "An Ethereum node location must be provided along with the name and capabilities (optional). \
+                         Try e.g. 'mainnet:URL'."
+                );
+            })?;
 
-                if loc.is_empty() {
-                    return Err(format_err!("Ethereum node URL cannot be an empty string"));
+            let (network_capabilities_str, url) = rest.split_at(http_split_at);
+            let network_capabilities: Vec<NetworkCapability> = match network_capabilities_str.len()
+            {
+                0 => vec![NetworkCapability::Archive, NetworkCapability::Traces],
+                length => {
+                    network_capabilities_str[..length - 1].split(',')
+                        .map(|capability_str| {
+                            Ok(NetworkCapability::from_str(dbg!(capability_str))?)
```

Comment:
Drop the `dbg!`

Reply:
One cool trick I think you can do with `FromStr` impls is to just call `capability_str.parse()?` and it'll automatically coerce the type into `NetworkCapability`.